### PR TITLE
Fix OOM killer memory measurement

### PR DESCRIPTION
**Pull request by bduffany**
Before, we computed memory usage fraction as `max(0, schedulable_memory - sys_free_memory) / schedulable_memory` which might never quite reach 0.95 (the OOM kill threshold) if the system has enough free memory outside of the executor cgroup.

This PR changes the calculation to:
- If the current cgroup has an effective memory limit (i.e. `memory.max` is set on either the current cgroup or an ancestor cgroup), then use `(cgroup_current_memory - cgroup_inactive_file_memory) / cgroup_effective_memory_limit`
- Otherwise, use `used_memory / total_memory`

This `cgroup_current_memory - cgroup_inactive_file_memory` calculation is inspired by the k8s "working set" memory calculation which is what it uses to determine whether a node is low on memory before it starts evicting pods. Inactive page cache is reclaimable under memory pressure, so it shouldn't count toward usage. We subtract `inactive_file` specifically, rather than the whole `file` stat, because `file` also includes tmpfs/shmem, which can't be reclaimed (tmpfs has no backing file and can only be swapped, and executors run without swap), and because active page cache is the hot set that the kernel keeps under pressure, so it counts as used.

Edge case: on cgroup v1, the executor can't determine its cgroup (only v2 is supported), so it logs a warning and uses the system memory calculation.

This PR also introduces integration tests that run under firecracker. I figure it's worth having these tests because kernel memory management is pretty complex and the cost of getting this behavior wrong is pretty high (we could be over-killing tasks or under-killing tasks)

### CODE REVIEW

File: enterprise/server/cmd/executor/executor.go [hunk 1]
```diff
@@ -387,22 +398,26 @@ func main() {
...
-	tasksCgroupParent, err := setupCgroups()
+	cgroups, err := setupCgroups()
 	if err != nil {
 		log.Fatalf("cgroup setup failed: %s", err)
 	}
 
 	var oomKiller oomkiller.Killer
 	if oomkiller.Enabled() {
-		k, err := oomkiller.New(rootContext, oomkiller.NewMemoryMonitor())
+		monitor, err := oomkiller.NewMemoryMonitor(cgroups.StartingCgroup)
```
[the remark this answers is not in the document]
Comment (bduffany) [reply enterprise/server/cmd/executor/executor.go:408]: eh I'd rather crash loop rather than have the OOM killer function incorrectly. Falling back to the system memory usage is not really a graceful fallback, it's just straight up incorrect.

**Review by vadimberezniker — APPROVED** [review]
